### ISSUE TITLE
Semantic problems when the kcl.mod file locates the location of the package and is nested

### ISSUE THREAD

**Issue by orangebees** [KusionStack/KCLVM#223]
## Enhancement
I'm the author of the glcc kcl package manager kpm and I'm having a problem building dependencies.
The dependency placement scheme I use is to place dependencies in external directories and implement dependencies through directory nesting.
If a depends on b and b depends on c, then

**a/main.k**
```
import external.b
hello = b.hellob
```
**b/main.k**
```
import external.c
hellob = c.helloc
```

**c/main.k**
```
helloc = "hello world"
```
The directory relationship is
```
.└── a
    ├── external
    │   └── b
    │       ├── external
    │       │   └── c
    │       │       ├── kcl.mod
    │       │       └── main.k
    │       ├── kcl.mod
    │       └── main.k
    ├── kcl.mod
    └── main.k
```

I learned from reading the documentation that kcl.mod plays the role of locating the root directory where the current package is located. In theory, my dependency placement method can work, but this is not the case. When the following modifications are made, it can work normally.
**b/main.k**
```
import external.b .external.c
hellob=c.helloc
```
So can the official support this semantics?

**Comment by chai2010** [comment]
the external dir can not include external sub dir. the kcl pkg manager should get all the depend pkgs, and put them to the external dir. for example:

```
+── a
 +── external
    │   └── b
    │      ├── kcl.mod
    │      └── main.k
    │    └── c
    │       ├── kcl.mod
    │       └── main.k
    ├── kcl.mod
    └── main.k
```

if the a/main.k import `b`， `b` import `c`, the pkg manager should put the `b` and `c` pkg to the `external/b` and `external/c` path.

**Comment by Peefy** [comment]
Closed by #466
